packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -0,0 +1,205 @@
+        '\n\n'
+        'This flag is set to true by default.'
+        '\n\n'
+        'When synthetic-package is set to false, it will generate the '
+        'localizations files in the directory specified by arb-dir by default. '
+        '\n\n'
+        'If output-dir is specified, files will be generated there.',
+    );
+    argParser.addOption(
+      'project-dir',
+      valueHelp: 'absolute/path/to/flutter/project',
+      help: 'When specified, the tool uses the path passed into this option '
+        'as the directory of the root Flutter project.'
+        '\n\n'
+        'When null, the relative path to the present working directory will be used.'
+    );
+  }
+
+  FileSystem _fileSystem;
+
+  @override
+  String get description => 'Generate localizations for the Flutter project.';
+
+  @override
+  String get name => 'generate_localizations';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+
+    precacheLanguageAndRegionTags();
+
+    final String inputPathString = argResults['arb-dir'] as String;
+    final String outputPathString = argResults['output-dir'] as String;
+    final String outputFileString = argResults['output-localization-file'] as String;
+    final String templateArbFileName = argResults['template-arb-file'] as String;
+    final String untranslatedMessagesFile = argResults['untranslated-messages-file'] as String;
+    final String classNameString = argResults['output-class'] as String;
+    final String preferredSupportedLocaleString = argResults['preferred-supported-locales'] as String;
+    final String headerString = argResults['header'] as String;
+    final String headerFile = argResults['header-file'] as String;
+    final bool useDeferredLoading = argResults['use-deferred-loading'] as bool;
+    final String inputsAndOutputsListPath = argResults['gen-inputs-and-outputs-list'] as String;
+    final bool useSyntheticPackage = argResults['synthetic-package'] as bool;
+    final String projectPathString = argResults['project-dir'] as String;
+
+    final LocalizationsGenerator localizationsGenerator = LocalizationsGenerator(_fileSystem);
+
+    try {
+      localizationsGenerator
+        ..initialize(
+          inputPathString: inputPathString,
+          outputPathString: outputPathString,
+          templateArbFileName: templateArbFileName,
+          outputFileString: outputFileString,
+          classNameString: classNameString,
+          preferredSupportedLocaleString: preferredSupportedLocaleString,
+          headerString: headerString,
+          headerFile: headerFile,
+          useDeferredLoading: useDeferredLoading,
+          inputsAndOutputsListPath: inputsAndOutputsListPath,
+          useSyntheticPackage: useSyntheticPackage,
+          projectPathString: projectPathString,
+        )
+        ..loadResources()
+        ..writeOutputFiles()
+        ..outputUnimplementedMessages(untranslatedMessagesFile, globals.logger);
+    } on FileSystemException catch (e) {
```

Comment:
consider not catching all file system exceptions: some of these might be programming errors in the tool.

Reply:
I've converted any `FileSystemException`s I threw in the tool into `L10nException`s and then removed the code catching file system exceptions